tests/ui/issues/issue-107518.stderr

```diff
@@ -0,0 +1,10 @@
+error: expected one of `.`, `?`, `]`, or an operator, found `;`
+  --> $DIR/issue-107518.rs:2:40
+   |
+LL |     (a_function_that_takes_an_array[0; 10]);
+   |                                        ^^ expected one of `.`, `?`, `]`, or an operator
```

Comment:
```suggestion
   |                                      ^ expected one of `.`, `?`, `]`, or an operator
```

The span is not quite correct. We shouldn't point at the literal and claim that we expected a dot, question etc. since that would mean we expected strings like `[0; .]`, `[0; ?]` etc. Instead, we should highlight the semicolon just like before.